Parser/spark.py

```diff
@@ -660,8 +665,8 @@ def ambiguity(self, rules):
             sortlist.append((len(rhs), name))
             name2index[name] = i
         sortlist.sort()
-        list = map(lambda (a,b): b, sortlist)
-        return rules[name2index[self.resolve(list)]]
+        mapped = list(map(lambda ab: ab[1], sortlist))
```

Comment:
mapped = [b for a, b in sortlist]

Reply:
Fixed in ebea5cdeac637c43c87cf03bb513bf9050a81eca.